### PR TITLE
 Update Eclipse 2024-06

### PR DESCRIPTION
**Pull request by Mailaender**
https://github.com/snapcrafters/eclipse/actions/workflows/sync-upstream.yml is disabled.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: eclipse
-base: core22
+base: core24
```
Comment (jnsgruk) [inline snap/snapcraft.yaml:2]: Just to confirm, have you tested this works as expected? I've not seen many desktop apps on core24 yet?

**Review by jnsgruk — APPROVED** [review]
One minor question, but otherwise LGTM

File: snap/snapcraft.yaml (hunk 1 above)
Comment (jnsgruk) [reply snap/snapcraft.yaml:2]: Okay, got it working on a fresh box, let me merge and see if the automated screenshotting all works!

Comment (Mailaender) [reply snap/snapcraft.yaml:2]: Sorry, I can't test locally due to https://bugs.launchpad.net/snapcraft/+bug/2069292

**Pull request merged**